### PR TITLE
Interview process

### PR DESCRIPTION
**Pull request by CheezItMan**
Basic translation of the original [Interview introduction lesson](https://github.com/gSchool/ada-cs-fun-class/tree/master/08-interview-process) to Core.

### CODE REVIEW

File: interview-process/how-to-interview.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+### Determine Test Input-Output
+
+The next step is to determine sample input into the problem and the resulting output.  This allows the applicant to verify how well they understand the problem and identify how the answer is determined.
+
+When creating sample input-output look for edge-cases.  Solicit feedback from the interviewer to verify that the sample input and output values are correct and you understand the problem.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:108]:
```suggestion
When creating sample input-output, look for edge-cases.  Solicit feedback from the interviewer to verify that the sample input and output values are correct and that you understand the problem.
```

File: interview-process/how-to-interview.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+  - num = num mod 3
+
+loop until num mod 5 is not 0
+  - num = num mod 5
+
+if num is 1 return True
+Otherwise return False
+```
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:152]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+### !callout-success
+
+## Pro-Tip Use "We"
+
+When discussing the problem a candidate should try to engage the interviewer in the process as a collaborator.  Use terms like "we" when talking abou the problem and solution.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:157]:
```suggestion
When discussing the problem a candidate should try to engage the interviewer in the process as a collaborator.  Use terms like "we" when talking about the problem and the solution.  
```

File: interview-process/how-to-interview.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+```
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
+### !callout-success
+
+## Pro-Tip Use "We"
+
+When discussing the problem a candidate should try to engage the interviewer in the process as a collaborator.  Use terms like "we" when talking abou the problem and solution.  
+
+In general, an interviewer *wants* the candidate to succeed and by engaging in a collaborative process, you can both become more at-ease and solicit more helpful feedback.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:159]:
```suggestion
In general, an interviewer *wants* the candidate to succeed. By engaging in a collaborative process, you can both become more at-ease and solicit more helpful feedback.
```

File: interview-process/how-to-interview.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+
+### !end-callout
+
+### Begin Coding
+
+Once the candidate has discussed the approach with the interviewer and answered any questions it is time to begin coding.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:165]:
```suggestion
Once the candidate has discussed the approach with the interviewer and answered any questions, it is time to begin coding.
```

File: interview-process/how-to-interview.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+
+While coding the solution it is important to remember to:
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:167]:
```suggestion
While coding the solution, it is important to remember to:
```

File: interview-process/how-to-interview.md [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+
+- Explain each step while coding
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:169]:
```suggestion
- Explain each step as you code
```

File: interview-process/how-to-interview.md [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+- When pausing to think, speak aloud to explain the thought process, interviewers need to understand what the candidate is thinking
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:170]:
```suggestion
- When pausing to think, speak out loud about your thought process. Interviewers need to understand what the candidate is thinking.
```

File: interview-process/how-to-interview.md [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+- Pause regularly to test each step using the input-output values from before
+
+Remember **communication is key**.  An interviewer needs to understand the candidate's thinking.
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:175]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+### !callout-success
+
+## Pro Tip - Ask for Help
+
+When stuck, there is nothing wrong with a candidate asking for help.  The interviewer could respond with, "no," but often will respond with a useful tip to get the candidate past the sticking point. 
+
+### !end-callout
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:185]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 11]
```diff
@@ -0,0 +1,327 @@
...
+
+  def test_six(self):
+    self.assertEqual(ugly_number(150), True)
+  def test_151(self):
+      self.assertEqual(ugly_number(151), False)
+```
+
+##### !end-tests
+
+<!-- other optional sections -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:279]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 12]
```diff
@@ -0,0 +1,327 @@
...
+for divisor in [2, 3, 5]:
+    while num % divisor == 0:
+        num = num // divisor
+
+return num == 1
+```
+
+##### !end-hint
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+<!-- !explanation - !end-explanation (markdown, students can see after answering correctly) -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:294]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:298]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 14]
```diff
@@ -0,0 +1,327 @@
...
+
+### Test Your Solution
+
+When a candidate finishes programming they should test their solution with the sample input and output values and verify that the solution works.  Testing helps verify that no edge-case has been missed.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:302]:
```suggestion
When a candidate finishes programming, they should test their solution with the sample input and output values and verify that the solution works.  Testing helps verify that no edge-cases have been missed.
```

File: interview-process/how-to-interview.md [hunk 15]
```diff
@@ -0,0 +1,327 @@
...
+
+### Solution Discussion
+
+After presenting the solution, the candidate will often be expected to discuss the pros and cons of the solution as written.  Candidates should be prepared to identify the time and space complexity and talk about alternate approaches, if time allows.  Interviews can be stressful and time constrained, so a candidate may not have had time to refactor and create an ideal solution, but they can discuss what next steps they would follow.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:306]:
```suggestion
After presenting the solution, the candidate will often be expected to discuss the pros and cons of the solution as it is written.  Candidates should be prepared to identify the time and space complexity and talk about alternate approaches (if time allows).  

Interviews can be stressful and time constrained, so a candidate may not have had time to refactor and create an ideal solution, but they can discuss what next steps they would follow.
```

File: interview-process/how-to-interview.md [hunk 16]
```diff
@@ -0,0 +1,327 @@
...
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:308]:
```suggestion
```

File: interview-process/how-to-interview.md [hunk 17]
```diff
@@ -0,0 +1,327 @@
...
+### !callout-success
+
+## Perfection is **NOT** the Goal
+
+The goal of a good interviewer is not to see the candidate arrive at a perfect solution, even if one exists.  A perfect solution is nice, but the real goal is to understand what the candidate could bring to the team.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:313]:
```suggestion
The goal of a good interviewer is to understand what the candidate could bring to the team. It is **not** to see if the candidate arrives at a perfect solution, whether or not one might exist. 
```

File: interview-process/how-to-interview.md [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+- Can this candidate communicate their thoughts in a technical environment?
+- Would this candidate be a good team member?
+- How well does this candidate solve problems?
+- Is this a person who could mesh with our team?
+
+### !end-callout
+
+### Interview Conclusion
+
+At the conclusion of the interview the candidate should thank the interviewer for their time.  The company will normally follow up later to either move forward or not.
```
Comment (spitsfire) [inline interview-process/how-to-interview.md:327]:
```suggestion
At the conclusion of the interview, the candidate should thank the interviewer for their time.  The company will normally follow up later to either move forward or not.
```

File: interview-process/interview-introduction.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+  - Coding Challenge
+  - Phone Screening
+
+## Vocabulary and Synonyms
+
+| Vocab	|  Definition	| Synonym | How to Use in a Sentence |
+|---	|---	|---	|---	|
+|  Whiteboard Interview  	|  A problem solving exercise where a job applicant is asked to solve a programming or design problem on a whiteboard. 	|    Technical Interview	|   I have a whiteboard interview next week, so I need a buddy to practice with. 	|
+| Phone Screen | An initial telephone or internet chat with a job candidate to ensure the candidate's suitability for an in-peron interview.  This could involve a hiring manager, but is more likely to involve a recruiter. | &nbsp; | I have a phone screening with a recruiter at Dynamic Widgets in 3 hours. | 
+| Coding Challenge | A "homework" assignment given to job candidates to assess their programming skills and practices such as regularly committing and writing tests. |  | I'm trying to write tests for my Dynamic Widgets coding challenge assignment.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:23]:
```suggestion
| Coding Challenge | A "homework" assignment given to job candidates to assess their programming skills and practices, such as regularly committing and writing tests. |  | I'm trying to write tests for my Dynamic Widgets coding challenge assignment. 
```

File: interview-process/interview-introduction.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+
+When applying for a junior developer position in the United States, applicants typically follow these steps.
+
+1. Applying for a position
+1. Phone or video chat screening
+1. In-person or video interview
+1. Offer & Negotiation
+1. Acceptance
+
+These steps are typical in the process, but the process can vary greatly.  Companies may not engage in the phone screening or might require additional interviews, or provide homework in the form of a coding challenge.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:35]:
```suggestion
These steps are typical in the process, but the process can vary greatly.  Companies may not engage in phone screening, or they may require additional interviews. They may provide homework in the form of a coding challenge as well.
```

File: interview-process/interview-introduction.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+### Application
+
+A junior developer can apply for a position in many ways, this can include an online application form on the company website or a job-listing site like [Indeed](https://www.indeed.com/q-Apply-jobs.html).  Sometimes it can also include talking to a recruiter, a colleague at the company, or a hiring manager at a local meet-up.  Universities like the University of Washington hold job-fairs.  There are a lot of ways to get in the door as a junior developer, but the most effective strategies typically involve networking.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:39]:
```suggestion
A junior developer can apply for a position in many ways. This can include an online application form on the company website or a job-listing site, like [Indeed](https://www.indeed.com/q-Apply-jobs.html).  Sometimes it can also include talking to a recruiter, a colleague at the company, or a hiring manager at a local meet-up.  Universities like the University of Washington hold job-fairs.  

There are a lot of ways to get in the door as a junior developer, but the most effective strategies typically involve networking.  
```

File: interview-process/interview-introduction.md [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+1. Offer & Negotiation
+1. Acceptance
+
+These steps are typical in the process, but the process can vary greatly.  Companies may not engage in the phone screening or might require additional interviews, or provide homework in the form of a coding challenge.
+
+### Application
+
+A junior developer can apply for a position in many ways, this can include an online application form on the company website or a job-listing site like [Indeed](https://www.indeed.com/q-Apply-jobs.html).  Sometimes it can also include talking to a recruiter, a colleague at the company, or a hiring manager at a local meet-up.  Universities like the University of Washington hold job-fairs.  There are a lot of ways to get in the door as a junior developer, but the most effective strategies typically involve networking.  
+
+When possible, candidates should take advantage of opportunities to meet with engineers, managers and recruiters either in person or electronically.  Online applications turned into hiring systems are much easier to ignore than a human being with a connection to the team or company.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:41]:
```suggestion
When possible, candidates should take advantage of opportunities to meet with engineers, managers, and recruiters, either in person or electronically.  Online applications turned into hiring systems are much easier to ignore than a human being with a connection to the team or company.
```

File: interview-process/interview-introduction.md [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+
+### Phone Screen
+
+A phone screen, despite the name might be either on the phone or via an online platform such as Skype or Google Hangout.  Candidates will likely initially talk to a recruiter, but the call may involve a hiring manager.  Candidates may or may not be asked technical questions, but the goal of a phone screen is generally to narrow down the list of candidates for on site interviews.  Generally the discussion centers on work experience, projects worked on and how the candidate's interests align with the company and team applied for.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:45]:
```suggestion
Despite the name, a phone screen can be either on the phone or via an online platform, such as Skype or Google Hangout.  Candidates will likely talk to a recruiter initially, but the call may involve a hiring manager.  

Candidates may or may not be asked technical questions, but the goal of a phone screen is to narrow down the list of candidates for onsite interviews.  Generally, the discussion centers on work experience, projects worked on, and how the candidate's interests align with the company and team applied for.
```

File: interview-process/interview-introduction.md [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+
+Before a phone screen it is a good idea to:
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:47]:
```suggestion
Before a phone screen, it is a good idea to:
```

File: interview-process/interview-introduction.md [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+
+- Review the job listing and take notes
+- Review the resume as the interviewer will likely ask about work experience
+- Prepare some questions to help determine how attractive the position
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:51]:
```suggestion
- Prepare some questions to help determine how attractive the position is
```

File: interview-process/interview-introduction.md [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+- Find a quiet place with good connectivity.
+
+### In-person Interview
+
+A technical in-person interview can vary greatly in length and process.  Typically however a candidate will have some initial time to greet and introduce yourself to the interviewer(s) with some brief chitchat.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:56]:
```suggestion
A technical in-person interview can vary greatly in length and process; however, typically a candidate will have some initial time to greet and introduce yourself to the interviewer(s) with some brief chitchat.  
```

File: interview-process/interview-introduction.md [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+- Review the job listing and take notes
+- Review the resume as the interviewer will likely ask about work experience
+- Prepare some questions to help determine how attractive the position
+- Find a quiet place with good connectivity.
+
+### In-person Interview
+
+A technical in-person interview can vary greatly in length and process.  Typically however a candidate will have some initial time to greet and introduce yourself to the interviewer(s) with some brief chitchat.  
+
+Then the interviewer will begin the technical portion of the interview.  This is often a _whiteboard interview_, meant to test the candidate's ability to solve new problems and communicate their thoughts.  Candidates are traditionally given a problem and a whiteboard and asked to develop a solution and explain it to the interviewer.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:58]:
```suggestion
Then the interviewer will begin the technical portion of the interview.  This is often a _whiteboard interview_, meant to test the candidate's ability to solve new problems and communicate their thoughts.  Candidates are traditionally given a problem and a whiteboard, asked to develop a solution, and then explain it to the interviewer.
```

File: interview-process/interview-introduction.md [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+
+In recent times, candidates are often interviewed in an online technical interview using Zoom, Meet, Cisco WebEx and a shared online code editor such as [Feplit](https://replit.com/), [Coderpad](https://coderpad.io/), [HackerRank](https://hackerrank.com/) or [VS Code LiveShare](https://docs.microsoft.com/en-us/visualstudio/liveshare/use/vscode).  In a process similar to an in-person interview candidates are presented with a programming problem and work through a solution live with the interviewers demonstrating problem solving, and communication skills.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:60]:
```suggestion
In recent times, candidates are often interviewed online, using Zoom, Meet, Cisco WebEx, and use a shared online code editor, such as [Feplit](https://replit.com/), [Coderpad](https://coderpad.io/), [HackerRank](https://hackerrank.com/), or [VS Code LiveShare](https://docs.microsoft.com/en-us/visualstudio/liveshare/use/vscode).  In a process similar to an in-person interview, candidates are presented with a programming problem and are asked to work through a solution live with the interviewers, demonstrating problem solving and communication skills.
```

File: interview-process/interview-introduction.md [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:62]:
```suggestion
```

File: interview-process/interview-introduction.md [hunk 30]
```diff
@@ -0,0 +1,101 @@
...
+### !callout-success
+
+## Mock Interviews
+
+As an Ada student you will be matched up for a practice interview, called a "mock interview" with an experienced hiring manager or engineer.  They will give you a sample interview problem and conduct a practice interview with you.  After the "mock interview" the interviewer will write up a set of feedback for you to help you prepare for your actual internship interviews.
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:67]:
```suggestion
As an Ada student, you will be matched up for a practice interview called a "mock interview" with an experienced hiring manager or engineer.  They will give you a sample interview problem and conduct a practice interview with you.  After the "mock interview," the interviewer will write up feedback for you to help you prepare for your actual internship interviews.
```

File: interview-process/interview-introduction.md [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+Remember:  
+
+![don't panic](/assets/interview-process_interview-introduction_dont-panic.jpg)
+
+The mock interview is designed to give you practice and experience with a technical interview and provide you useful feedback.
+
+### !end-callout
+
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
```
Comment (spitsfire) [inline interview-process/interview-introduction.md:78]:
```suggestion
```

File: interview-process/interview-recording.md [hunk 32]
```diff
@@ -0,0 +1,46 @@
...
+2. Given any integer, print an English phrase that describes the integer (e.g. One Thousand Two Hundred Thirty Four).
+-->
+
+### Partner 2 (choose to ask one of the two)
+
+View markdown when you are ready to ask
+<!--
+3. Given two strings, write a method to decide if one is a permutation of the other (different ordering of the same letters)?
+4. Given two lists of unordered numbers, find the intersection of the two lists (the numbers that are in common).
+-->
```
Comment (spitsfire) [inline interview-process/interview-recording.md:31]: are these supposed to be commented out?

File: interview-process/interview-recording.md [hunk 33]
```diff
@@ -0,0 +1,46 @@
...
+
+## Examples of things you are looking for in your recorded playback
+
+- Is your writing legible?
+- Do you block the write board when you talk?
```
Comment (spitsfire) [inline interview-process/interview-recording.md:36]:
```suggestion
- Do you block the whiteboard when you talk?
```

File: interview-process/interview-recording.md [hunk 34]
```diff
@@ -0,0 +1,46 @@
...
+- Do you have long pauses? Or are they shorter in playback than they are in your head?
+- Do you appear confident?
+- How is your eye contact and demeanor?
+- Do you mumble?
+- Do you talk to the whiteboard or the interviewer?
+- Are you unnecessarily apologizing?
+- Are you saying things like "I don't know where this is going" instead of something like "I might rethink this later, but I'll start here"?
```
Comment (spitsfire) [inline interview-process/interview-recording.md:43]:
```suggestion
- Are you saying things like, "I don't know where this is going" instead of something like, "I might rethink this later, but I'll start here"?
```

File: interview-process/interview-recording.md [hunk 35]
```diff
@@ -0,0 +1,46 @@
...
+- Are you using technical describing words appropriately or are you saying a lot of "this", "that", etc?
```
Comment (spitsfire) [inline interview-process/interview-recording.md:44]:
```suggestion
- Are you using technical-describing words appropriately, or are you saying a lot of "this", "that", etc?
```

**Review by spitsfire — CHANGES_REQUESTED** [review]
One thing to mention: this goes from second person to third and back a lot. I did my best to at least make it consistent within each sentence, even if I didn't select one throughout the entire PR
